src/main/java/NestedNullRefactoring.java

```diff
@@ -0,0 +1,151 @@
+
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.List;
+
+import org.eclipse.jdt.core.dom.AST;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.Block;
+import org.eclipse.jdt.core.dom.Expression;
+import org.eclipse.jdt.core.dom.IMethodBinding;
+import org.eclipse.jdt.core.dom.InfixExpression;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+import org.eclipse.jdt.core.dom.MethodInvocation;
+import org.eclipse.jdt.core.dom.NullLiteral;
+import org.eclipse.jdt.core.dom.ParenthesizedExpression;
+import org.eclipse.jdt.core.dom.PrefixExpression;
+import org.eclipse.jdt.core.dom.PrimitiveType;
+import org.eclipse.jdt.core.dom.ReturnStatement;
+import org.eclipse.jdt.core.dom.SimpleName;
+import org.eclipse.jdt.core.dom.Statement;
+import org.eclipse.jdt.core.dom.Type;
+import org.eclipse.jdt.core.dom.rewrite.ASTRewrite;
+
+public class NestedNullRefactoring extends Refactoring {
+	public static final String NAME = "NestedNullRefactoring";
+
+	private final Dictionary<IMethodBinding, Expression> applicableMethods;
+
+	public NestedNullRefactoring() {
+		applicableMethods = new Hashtable<>();
+	}
+
+	@Override
+	public boolean isApplicable(ASTNode node) {
+
+		if (node instanceof MethodInvocation invocation) {
+			return isApplicable(invocation);
+		}
+
+		// Confirm that the ASTNode is a method
+		if (node instanceof MethodDeclaration declaration) {
+			return isApplicable(declaration);
+		}
+
+		return false;
+
+	}
+
+	/*
+	 * Returns true iff the provided invocation is of a registered one-line method
+	 * that returns the result of a null check
+	 */
+	public boolean isApplicable(MethodInvocation invocation) {
+		// Check if Method Invocation is in applicableMethods
+		if (applicableMethods.get(invocation.resolveMethodBinding()) != null) {
+			System.out.println("[DEBUG] Invocation of applicable method found");
+			return true;
+		}
+		return false;
+	}
+
+	/*
+	 * Returns true iff Node is a one-line method that returns the result of a null
+	 * check
+	 */
+	public boolean isApplicable(MethodDeclaration declaration) {
+		// Confirm that the method returns a boolean
+		Type retType = declaration.getReturnType2();
+		boolean isBooleanDeclaration = (retType.isPrimitiveType()
+				&& ((PrimitiveType) retType).getPrimitiveTypeCode() == PrimitiveType.BOOLEAN);
+		if (!(isBooleanDeclaration)) {
+			return false;
+		}
+
+		// Checks if there are any parameters
+		// TODO: Make work with Parameters
```

Comment:
What's wrong with the current implementation? I don't think we need to handle cases with parameters.

Reply:
The right way to address a comment like this one is in the code (i.e., by adding your answer to the comment) rather than in the code review tool - only I will read it here, but all future code readers who have the same question will benefit if you answer in the code itself. (This is a general policy that you should try to adopt when responding to review comments: usually, they should elicit some change to the code itself rather than a response in the review tool.)